src/RADCodeGen.hs

```diff
@@ -0,0 +1,557 @@
+module RADCodeGen where
+  import Grammar
+  import LALR
+  import GenUtils (mapDollarDollar)
+  import RADTools (XGrammar(..), showItem, showProd, lhs, showRecognitionPoint, recognitionPoints, rhsAfterDot)
+  import RADStateGen
+  import Control.Monad
+  import Data.List
+  import Data.Maybe
+  import Data.Text (pack, unpack, replace)
+  import GHC.Arr ((!), indices)
+  
+
+  data ParserType = Normal | Monad | MonadLexer deriving (Eq, Show)
+  
+  data GenOptions = GenOptions {
+    ptype :: ParserType,
+    
+    wrapperType :: String, -- e.g. "Parser"
+    errorTokenType :: String, -- e.g. "ErrorToken"
```

Comment:
Is this something that needs configuring? What do the other Backends do?

Reply:
No, this doesn't really need to be configured, we just need to choose some string for the dummy type. We could also fully remove the ErrorToken dummy type; this would change the number of arguments of affected continuations and semantic actions (so we would have to be more careful at other places), but should still work.